46.md

```diff
@@ -2,161 +2,67 @@ NIP-46
-
-- `relay` URL of the relay of choice where the **App** is connected and the **Signer** must send and listen for messages.
-- `metadata`  metadata JSON of the **App** 
-    - `name` human-readable name of the **App** 
-    - `url` (optional) URL of the website requesting the connection
-    - `description` (optional) description of the **App**
-    - `icons` (optional) array of URLs for icons of the **App**.
-
-#### JavaScript
-
-```js
-const uri = `nostrconnect://<pubkey>?relay=${encodeURIComponent("wss://relay.damus.io")}&metadata=${encodeURIComponent(JSON.stringify({"name": "Example"}))}`
-```
-
-#### Example
-```sh
-nostrconnect://b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.damus.io&metadata=%7B%22name%22%3A%22Example%22%7D
-```
-
-
-
-## Flows
-
-The `content` field contains encrypted message as specified by [NIP04](https://github.com/nostr-protocol/nips/blob/master/04.md). The `kind` chosen is `24133`.
-
-### Connect
-
-1. User clicks on **"Connect"** button on a website or scan it with a QR code
-2. It will show an URI to open a "nostr connect" enabled **Signer** 
-3. In the URI there is a pubkey of the **App** ie. `nostrconnect://<pubkey>&relay=<relay>&metadata=<metadata>`
-4. The **Signer** will send a message to ACK the `connect` request, along with his public key
-
-### Disconnect (from App)
-
-1. User clicks on **"Disconnect"** button on the **App**
-2. The **App** will send a message to the **Signer** with a `disconnect` request
-3. The **Signer** will send a message to ACK the `disconnect` request
-
-### Disconnect (from Signer)
-
-1. User clicks on **"Disconnect"** button on the **Signer**
-2. The **Signer** will send a message to the **App** with a `disconnect` request
-
-
-### Get Public Key
-
-1. The **App** will send a message to the **Signer** with a `get_public_key` request
-3. The **Signer** will send back a message with the public key as a response to the `get_public_key` request
-
-### Sign Event
-
-1. The **App** will send a message to the **Signer** with a `sign_event` request along with the **event** to be signed
-2. The **Signer** will show a popup to the user to inspect the event and sign it
-3. The **Signer** will send back a message with the event including the `id` and the schnorr `signature` as a response to the `sign_event` request
-
-### Delegate
-
-1. The **App** will send a message with metadata to the **Signer** with a `delegate` request along with the **conditions** query string and the **pubkey** of the **App** to be delegated.
-2. The **Signer** will show a popup to the user to delegate the **App** to sign on his behalf
-3. The **Signer** will send back a message with the signed [NIP-26 delegation token](https://github.com/nostr-protocol/nips/blob/master/26.md) or reject it
-
-
+  - params: [`pubkey`, `nip4 ciphertext`]
+  - result: [`plaintext`]
```

Comment:
since we are revamping this NIP, can we change nip04_decrypt result to not be an array? its the only result that returns an array, was there a reason for that? seems inconsistent.

Reply:
No, because it would break implementations. This rewrite doesn't try to change the functionality.

Reply:
ok fine. I guess nip4 will soon be deprecated in favor of nip44 soon anyway.